### PR TITLE
Adds getting started tutorial.

### PR DESCRIPTION
**Pull request by ashleyrback**
Partially fulfills #78.

Tutorial covers:
 * `Spectra` creation
 * Filling spectrum
 * Plotting spectrum
 * Convolution and cuts
 * Other manipulations e.g. `shrink_to_roi`, `rebin` and `scale`

The tutorial is created in an ipython notebook, and also as a separate script in the `scripts` directory and on the wiki.

### CODE REVIEW

**Comment by ashleyrback** [comment]
@arushanova there are details on the [wiki]( https://github.com/snoplusuk/echidna/wiki/GettingStarted) explaining how to run the tutorial with ipython notebook.

**Comment by arushanova** [comment]
Thank you Ashley, this is a very nice tutorial. I just have a couple of comments for getting_started.ipynb. 
1) The plots are not shown in ipython. Please see the detailed comments bellow the lines. 
2) I think it will be more consistent to say either "we, our" or "you" throughout the tutorial.

File: echidna/scripts/tutorials/getting_started.py [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+
+import echidna.output.plot as plot
+import echidna.output.plot_root as plot_root
+
+
+# To plot the projection of the spectrum on the `energy_mc` axis:
+
+# In[13]:
+
+plot.plot_projection(spectrum_1, "energy_mc", fig_num=1)
```
Comment (arushanova) [inline echidna/scripts/tutorials/getting_started.py]: The plot didn't appear. 
<matplotlib.figure.Figure at 0x3942810>

File: echidna/scripts/tutorials/getting_started.py [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+
+
+# and to plot the projection on the `radial_mc` axis, this time using root:
+
+# In[15]:
+
+plot_root.plot_projection(spectrum_1, "radial_mc")
```
Comment (arushanova) [inline echidna/scripts/tutorials/getting_started.py]: This on the other hand worked, the plot appears.

File: echidna/scripts/tutorials/getting_started.py [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+    plt.ylim(ymin=0.0)
+    plt.xlabel(dimension + " [" + par.get_unit() + "]")
+    plt.ylabel("Events per " + str(width) + " " + par.get_unit() + " bin")
+    return fig
+
+
+# In[25]:
+
+fig1 = overlay_spectra(spectrum_1, smeared_spectrum, fignum=4)
+plt.show()
```
Comment (arushanova) [inline echidna/scripts/tutorials/getting_started.py]: The plot didn't appear. It was written:
ValueError                                Traceback (most recent call last)
<ipython-input-14-3378db744f5a> in <module>()
----> 1plot.plot_surface(spectrum_1, "energy_mc", "radial_mc")

/.data/snoplus/evelina/virtual_echidna/echidna/echidna/output/plot.pyc in plot_surface(spectra, dimension1, dimension2)
    192     """
    193     fig = plt.figure()
--> 194axis = fig.add_subplot(111, projection='3d')
    195     index1 = spectra.get_config().get_index(dimension1)
    196     index2 = spectra.get_config().get_index(dimension2)

/.data/snoplus/evelina/virtual_echidna/lib/python2.7/site-packages/matplotlib/figure.pyc in add_subplot(self, *args, **kwargs)
    944         else:
    945             projection_class, kwargs, key = process_projection_requirements(
--> 946                 self, *args, **kwargs)
    947 
    948             # try to find the axes with this key in the stack

/.data/snoplus/evelina/virtual_echidna/lib/python2.7/site-packages/matplotlib/projections/__init__.pyc in process_projection_requirements(figure, *args, **kwargs)
     96 
     97     if isinstance(projection, six.string_types) or projection is None:
---> 98projection_class = get_projection_class(projection)
     99     elif hasattr(projection, '_as_mpl_axes'):
    100         projection_class, extra_kwargs = projection._as_mpl_axes()

/.data/snoplus/evelina/virtual_echidna/lib/python2.7/site-packages/matplotlib/projections/__init__.pyc in get_projection_class(projection)
     65         return projection_registry.get_projection_class(projection)
     66     except KeyError:
---> 67         raise ValueError("Unknown projection '%s'" % projection)
     68 
     69 

ValueError: Unknown projection '3d'

Comment (arushanova) [inline echidna/scripts/tutorials/getting_started.py]: The plot didn't appear. This was written:
ValueError                                Traceback (most recent call last)
<ipython-input-14-3378db744f5a> in <module>()
----> 1plot.plot_surface(spectrum_1, "energy_mc", "radial_mc")

/.data/snoplus/evelina/virtual_echidna/echidna/echidna/output/plot.pyc in plot_surface(spectra, dimension1, dimension2)
    192     """
    193     fig = plt.figure()
--> 194axis = fig.add_subplot(111, projection='3d')
    195     index1 = spectra.get_config().get_index(dimension1)
    196     index2 = spectra.get_config().get_index(dimension2)

/.data/snoplus/evelina/virtual_echidna/lib/python2.7/site-packages/matplotlib/figure.pyc in add_subplot(self, *args, **kwargs)
    944         else:
    945             projection_class, kwargs, key = process_projection_requirements(
--> 946                 self, *args, **kwargs)
    947 
    948             # try to find the axes with this key in the stack

/.data/snoplus/evelina/virtual_echidna/lib/python2.7/site-packages/matplotlib/projections/__init__.pyc in process_projection_requirements(figure, *args, **kwargs)
     96 
     97     if isinstance(projection, six.string_types) or projection is None:
---> 98projection_class = get_projection_class(projection)
     99     elif hasattr(projection, '_as_mpl_axes'):
    100         projection_class, extra_kwargs = projection._as_mpl_axes()

/.data/snoplus/evelina/virtual_echidna/lib/python2.7/site-packages/matplotlib/projections/__init__.pyc in get_projection_class(projection)
     65         return projection_registry.get_projection_class(projection)
     66     except KeyError:
---> 67         raise ValueError("Unknown projection '%s'" % projection)
     68 
     69 

ValueError: Unknown projection '3d'

File: echidna/scripts/tutorials/getting_started.py [hunk 4]
```diff
@@ -0,0 +1,384 @@
...
+# does our final spectrum look like?
+
+# In[30]:
+
+print smeared_spectrum._data
+
+
+# In[32]:
+
+plot.plot_projection(smeared_spectrum, "energy_mc", fig_num=5)
```
Comment (arushanova) [inline echidna/scripts/tutorials/getting_started.py]: The plot didn't appear. There were no comments.

File: getting_started.ipynb [hunk 5]
```diff
@@ -0,0 +1,840 @@
...
+       "<matplotlib.figure.Figure at 0x7f8c08958050>"
+      ]
+     },
+     "execution_count": 13,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "plot.plot_projection(spectrum_1, \"energy_mc\", fig_num=1)"
```
Comment (arushanova) [inline getting_started.ipynb]: The plot didn't appear. This is what was written:
<matplotlib.figure.Figure at 0x3942810>

File: getting_started.ipynb [hunk 6]
```diff
@@ -0,0 +1,840 @@
...
+       "<ROOT.TH1D object (\"radial_mc\") at 0x402b130>"
+      ]
+     },
+     "execution_count": 15,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "plot_root.plot_projection(spectrum_1, \"radial_mc\")"
```
Comment (arushanova) [inline getting_started.ipynb]: This on the other hand worked, the plot appears.

File: getting_started.ipynb [hunk 7]
```diff
@@ -0,0 +1,840 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": 25,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "fig1 = overlay_spectra(spectrum_1, smeared_spectrum, fignum=4)\n",
+    "plt.show()"
```
Comment (arushanova) [inline getting_started.ipynb]: The plot didn't appear. This was written:
ValueError                                Traceback (most recent call last)
<ipython-input-14-3378db744f5a> in <module>()
----> 1plot.plot_surface(spectrum_1, "energy_mc", "radial_mc")

/.data/snoplus/evelina/virtual_echidna/echidna/echidna/output/plot.pyc in plot_surface(spectra, dimension1, dimension2)
    192     """
    193     fig = plt.figure()
--> 194axis = fig.add_subplot(111, projection='3d')
    195     index1 = spectra.get_config().get_index(dimension1)
    196     index2 = spectra.get_config().get_index(dimension2)

/.data/snoplus/evelina/virtual_echidna/lib/python2.7/site-packages/matplotlib/figure.pyc in add_subplot(self, *args, **kwargs)
    944         else:
    945             projection_class, kwargs, key = process_projection_requirements(
--> 946                 self, *args, **kwargs)
    947 
    948             # try to find the axes with this key in the stack

/.data/snoplus/evelina/virtual_echidna/lib/python2.7/site-packages/matplotlib/projections/__init__.pyc in process_projection_requirements(figure, *args, **kwargs)
     96 
     97     if isinstance(projection, six.string_types) or projection is None:
---> 98projection_class = get_projection_class(projection)
     99     elif hasattr(projection, '_as_mpl_axes'):
    100         projection_class, extra_kwargs = projection._as_mpl_axes()

/.data/snoplus/evelina/virtual_echidna/lib/python2.7/site-packages/matplotlib/projections/__init__.pyc in get_projection_class(projection)
     65         return projection_registry.get_projection_class(projection)
     66     except KeyError:
---> 67         raise ValueError("Unknown projection '%s'" % projection)
     68 
     69 

ValueError: Unknown projection '3d'

File: getting_started.ipynb [hunk 8]
```diff
@@ -0,0 +1,840 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "plot.plot_projection(smeared_spectrum, \"energy_mc\", fig_num=5)"
```
Comment (arushanova) [inline getting_started.ipynb]: The plot didn't appear. There were no comments.

**Comment by arushanova** [comment]
Could you please add init to the getting_started.py, the documentation doesn't appear online

**Comment by ashleyrback** [comment]
@arushanova changes pushed but still need to solve segmentation violation problem. See comment inline.

**Comment by ashleyrback** [comment]
@arushanova the segmentation violation problem has now been fixed, so all should be in place for you to give it a final check-over, please.

**Comment by arushanova** [comment]
I've tried to build the document and could see the documentation online. These are the warnings I get when make the doc:
phinx-build -b html -d _build/doctrees   . html/docs
Running Sphinx v1.3b2
loading pickled environment... done
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 10 source files that are out of date
updating environment: 1 added, 10 changed, 0 removed
RAT: Libraries loaded.7%] echidna.core                                                                                         
reading sources... [100%] modules                                                                                              
/.data/snoplus/evelina/virtual_echidna/echidna/docs/echidna.scripts.tutorials.rst:10: WARNING: autodoc: failed to import module u'echidna.scripts.tutorials.getting_started'; the following exception was raised:
Traceback (most recent call last):
  File "/.data/snoplus/evelina/virtual_echidna/lib/python2.7/site-packages/sphinx/ext/autodoc.py", line 378, in import_object
    __import__(self.modname)
  File "/data/snoplus/softwareSL6/root-5.34.30/lib/ROOT.py", line 352, in _importhook
    return _orig_ihook( name, *args, **kwds )
  File "/.data/snoplus/evelina/virtual_echidna/echidna/echidna/scripts/tutorials/getting_started.py", line 55, in <module>
    config = spectra.SpectraConfig.load_from_file("echidna/config/example.yml")
  File "/.data/snoplus/evelina/virtual_echidna/echidna/echidna/core/spectra.py", line 129, in load_from_file
    config = yaml.load(open(filename, 'r'))
IOError: [Errno 2] No such file or directory: 'echidna/config/example.yml'
looking for now-outdated files... none found
pickling environment... done
checking consistency... /.data/snoplus/evelina/virtual_echidna/echidna/docs/modules.rst:: WARNING: document isn't included in any toctree
done
preparing documents... done
writing output... [100%] modules                                                                                               
generating indices... genindex py-modindex
highlighting module code... [100%] echidna.scripts.multi_ntuple_spectrum                                                       
writing additional pages... search
copying static files... WARNING: html_static_path entry u'/.data/snoplus/evelina/virtual_echidna/echidna/docs/_static' does not exist
done
copying extra files... done
dumping search index in English (code: en) ... done
dumping object inventory... done
build succeeded, 3 warnings.

Build finished. The HTML pages are in html/docs.

**Comment by arushanova** [comment]
Here's pep8 output:
```
getting_started.py:87:1: E402 module level import not at top of file
getting_started.py:149:1: E402 module level import not at top of file
getting_started.py:150:1: E402 module level import not at top of file
getting_started.py:230:1: E402 module level import not at top of file
getting_started.py:265:1: E402 module level import not at top of file
getting_started.py:266:1: E402 module level import not at top of file
```

But I guess this is not a problem in this case?

**Comment by ashleyrback** [comment]
@arushanova, correct. In this case the module imports are part of the tutorial, so they are not all grouped at the top of the file. Well spotted though :+1:

**Comment by arushanova** [comment]
@ashleyrback The plots are nicely shown in ipython now.

**Comment by ashleyrback** [comment]
@arushanova, that should fix the docs.

**Comment by ashleyrback** [comment]
@arushanova, fixes plots displayed whilst building docs issue.

**Comment by arushanova** [comment]
merging!

**Pull request merged**